engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -522,34 +521,35 @@ public boolean isColorTerminalOutput() {
    */
   @TruffleBoundary
   public Object lookupJavaClass(String className) {
```

Comment:
Do these changes support multi-level nested classes as well? Reading the code, I can see that it just tries to replace the last occurence of `.` with `$` and resolve that symbol. This won't work for a class that is nested inside two classes. I cannot think of any use-case for that, but why was this change necessary again?

Reply:
I believe this code is easier to understand than the original one.

> tries to replace the last occurence of . with $ and resolve that symbol

Yes, and it repeats that action until there are any `.` left.

Reply:
> Scala classes that tend to use `$` encoding extensively. 

That's not a problem. This algorithm isn't seeking for `$`, but for `.`.

> Are anonymous classes somehow problematic?

We want to refer to inner classes via their _source name_ - e.g. with `.`, but their VM name contains `$`. By replacing the last `.` with `$` we check all possible VM names one after another.

> had some unit tests for this functionality.

We certainly have a lot of tests that require proper resolution of inner classes. If the algorithm was wrong, we'd see a CI failure.

> c&p of the Feature algorithm.

Originally I wanted to copy the original here in used algorithm to the feature, but then I realized it is needlessly complicated for what it is supposed to do. Thus I wrote new one and copied it here.